File: src/main/kotlin/ru/proshik/english/quizlet/telegramBot/controller/QuizletController.java
package ru.proshik.english.quizlet.telegramBot.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("redirect")
public class QuizletController {

    @GetMapping
    public void authRedirect(@RequestParam("state") String state,
                             @RequestParam("code") String code) {



    }

}

File: src/main/kotlin/ru/proshik/english/quizlet/telegramBot/controller/QuizletController.kt
package ru.proshik.english.quizlet.telegramBot.controller

import org.apache.log4j.Logger
import org.springframework.web.bind.annotation.*
import ru.proshik.english.quizlet.telegramBot.exception.AuthorizationException
import ru.proshik.english.quizlet.telegramBot.service.AuthorizationService
import javax.servlet.http.HttpServletResponse

@RestController
@RequestMapping("redirect")
class QuizletController(private val authorizationService: AuthorizationService) {

    companion object {

        private val LOG = Logger.getLogger(QuizletController::class.java)
    }

    @GetMapping
    fun authRedirect(@RequestParam("state") state: String,
                     @RequestParam("code") code: String,
                     httpServletResponse: HttpServletResponse) {

        authorizationService.authorization(state, code)

        httpServletResponse.setHeader("Location", "/")
        httpServletResponse.status = 302
    }

    @ExceptionHandler
    fun handleRuntimeException(ex: AuthorizationException) {
        LOG.warn("${ex.message}")
    }

    @ExceptionHandler
    fun handleRuntimeException(ex: Exception) {
        LOG.error("unexpected authorization exception", ex)
    }

}
